### PR TITLE
Reconstruct temperature in GRMHD and GHMHD

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Many thanks to @pajkosmi for doing most of the work on this.

Some plots of GRMHD blast wave.

Temperature reconstruction:

BlastWaveTempRecons_Temperature
BlastWaveTempRecons_RestMassDensity
BlastWaveTempRecons_Pressure

Pressure reconstruction:

BlastWavePressureRecons_Temperature
BlastWavePressureRecons_RestMassDensity
BlastWavePressureRecons_Pressure

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/PositivityPreservingAdaptiveOrder.hpp [hunk 1]
```diff
@@ -95,16 +95,16 @@ class PositivityPreservingAdaptiveOrderPrim : public Reconstructor {
     static constexpr Options::String help = {
         "The alpha parameter in the Persson convergence measurement. 4 is the "
         "right value, but anything in the range of 3-5 is 'reasonable'. "
-        "Smaller values allow for more oscillations. If specified to None, "
-        "then 7th-order reconstruction is not used."};
+        "Smaller values allow for more oscillations. If not specified then "
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/PositivityPreservingAdaptiveOrder.hpp]: I don't think you want this change... (undoes change by #5444)

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/PositivityPreservingAdaptiveOrder.hpp [hunk 2]
```diff
@@ -95,16 +95,16 @@ class PositivityPreservingAdaptiveOrderPrim : public Reconstructor {
...
+        "7th-order reconstruction is not used."};
   };
   struct Alpha9 {
     using type = Options::Auto<double, Options::AutoLabel::None>;
     static constexpr Options::String help = {
         "The alpha parameter in the Persson convergence measurement. 4 is the "
         "right value, but anything in the range of 3-5 is 'reasonable'. "
-        "Smaller values allow for more oscillations. If specified to None, "
-        "then 9th-order reconstruction is not used."};
+        "Smaller values allow for more oscillations. If not specified then "
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/PositivityPreservingAdaptiveOrder.hpp]: I don't think you want this change... (undoes change by https://github.com/sxs-collaboration/spectre/pull/5444)

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/ReconstructWork.tpp [hunk 3]
```diff
@@ -69,25 +69,34 @@ void compute_conservatives_for_reconstruction(
...
+  } else if constexpr (ThermodynamicDim == 2) {
     specific_internal_energy =
-        eos.specific_internal_energy_from_density(rest_mass_density);
-    temperature = eos.temperature_from_density(rest_mass_density);
+        eos.specific_internal_energy_from_density_and_temperature(
+            rest_mass_density, temperature);
+    pressure = eos.pressure_from_density_and_energy(rest_mass_density,
+                                                    specific_internal_energy);
+  } else if constexpr (ThermodynamicDim == 3) {
+    ERROR("1d, 2d, nor 3d EOS chosen");
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/ReconstructWork.tpp]: ?  I would have expected this in a last else.   Are you trying to say 3d not implemented yet?

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimitiveGhostData.hpp [hunk 4]
```diff
@@ -16,8 +16,8 @@ class Variables;
 namespace grmhd::ValenciaDivClean::subcell {
 /*!
  * \brief Computes the rest mass density \f$\rho\f$, electron fraction
- * \f$Y_e\f$, pressure \f$p\f$, Lorentz factor times the spatial velocity \f$W
- * v^i\f$, magnetic field \f$B^i\f$, and the divergence cleaning field
+ * \f$Y_e\f$, temperature \f$p\f$, Lorentz factor times the spatial velocity
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimitiveGhostData.hpp]: T for temperature?

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimitiveGhostData.hpp [hunk 5]
```diff
@@ -29,10 +29,11 @@ namespace grmhd::ValenciaDivClean::subcell {
  */
 class PrimitiveGhostVariables {
  private:
+  /// change pressure -> temperature here
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimitiveGhostData.hpp]: ?

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Rebased and squashed. Thanks for the review :)

**Review by kidder — APPROVED** [review]

**Pull request merged**